### ISSUE TITLE
s6-rc: fatal: timed out

### ISSUE THREAD

**Issue by usma0118** [pi-hole/docker-pi-hole#1186]
This is a: **Bug**  

## Details
I am running latest version on my k3s setup. it's throwing s6-rc error:

```
s6-rc: info: service cron: starting
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service cron successfully started
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service _startup: starting
Starting crond
fix-attrs: info: applying /etc/fix-attrs.d/01-resolver-resolv
fix-attrs: warning: fix-attrs is deprecated, please fix volume permissions in your container manager instead
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/05-changer-uid-gid.sh
 ::: Starting docker specific checks & setup for docker pihole/pihole
cont-init: info: /etc/cont-init.d/05-changer-uid-gid.sh exited 0
s6-rc: info: service legacy-cont-init successfully started

  [i] Installing configs from /etc/.pihole...
  [i] Existing dnsmasq.conf found... it is not a Pi-hole file, leaving alone!
  [✓] Installed /etc/dnsmasq.d/01-pihole.conf
  [✓] Installed /etc/dnsmasq.d/06-rfc6761.conf

  [i] Installing latest logrotate script...
        [i] Existing logrotate file found. No changes made.
s6-rc: fatal: timed out
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
prog: fatal: stopping the container.
s6-rc: info: service legacy-cont-init: stopping
s6-sudoc: fatal: unable to get exit status from server: Operation timed out
s6-rc: info: service cron: stopping
s6-rc: info: service legacy-cont-init successfully stopped
Stopping cron
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service cron successfully stopped
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

## How to reproduce the issue 

1. Environment data
  * Operating System: **Ubuntu 22**
  * Hardware: RasPi 4B
  * Kernel Architecture: aarch64
  * Docker Install Info and version: 
    - Software source: K3s
  * Hardware architecture: aarch64

2. docker-compose.yml contents, docker run shell command, or paste a screenshot of any UI based configuration of containers here
3. any additional info to help reproduce

## These common fixes didn't work for my issue
- [x] I have tried removing/destroying my container, and re-creating a new container
- [x] I have tried fresh volume data by backing up and moving/removing the old volume data
- [x] I have tried running the stock `docker run` example(s) in the readme (removing any customizations I added)
- [x] I have tried a newer or older version of Docker Pi-hole (depending what version the issue started in for me)
- [x] I have tried running without my volume data mounts to eliminate volumes as the cause

**Comment by usma0118** [comment]
same image runs fine on x86_64

**Comment by dschaper** [comment]
Which image are you using? Does this still happen with a previous image tag?

**Comment by Nimon77** [comment]
Hello, same problem here,
It works on my raspberry 4 (4Gb) but not on my raspberry 3 (1Gb), I'm in a swarm environment both `armv7l`, docker engine `20.10.17`
I have fallen back to tag `2022.07.1` temporarily

![image](https://user-images.githubusercontent.com/11821952/187042611-5e88caaa-e995-47fd-8a4a-b17f9c8e4d2d.png)

**Comment by usma0118** [comment]
I can also confirm, falling back to tag `2022.07.1 ` resolves the issue.

**Comment by joe-eklund** [comment]
> I can also confirm, falling back to tag `2022.07.1 ` resolves the issue.

Thanks. My DNS failed (same errors as OP) after I ran upgrades today, rolling back to this version brought it right back up.

**Comment by sleiner** [comment]
I have one more data point to add here: I'm getting the same error in a docker-compose based setup (without any env vars or additionally mounted config files). Interestingly enough, its always _the first start after container creation_ that fails.

```bash
# Assume the composition is down

docker-compose up  # will fail reproduibly (with same log as in the original ticket)

# ... as s6 exits, the container will not be running now

docker-compose up  # will succeed reproducibly
```

Notably, if I bring the composition down again (keeping volumes intact), the first `docker-compose up` will still fail again.

**Comment by casperklein** [comment]
Disclaimer: I am not very familiar with S6 overlay and not sure, if this applies to the current issue. But I wanted to mention it, in case it is.

I noticed several strange behaviors/errors the last days, when playing with the new `2022.08.2` image. They were not always reproducible, but in my tests they seem to happen more often on "slower" devices (compared to my workstation) like my RPI4.

Looking at the container logs I noticed, that services are started in parallel and even before `cont-init.d` scripts have finished. Race condition?

To ensure `cont-init.d` scripts are run before any service starts, I added "base" as dependency for all services. This [script](https://github.com/casperklein/homeassistant-addons-dev/blob/master/pi-hole/rootfs/bin/fix-dependencies.sh) is now run, while building my custom image. So far, that fixed it for me 🤞

**Comment by PromoFaux** [comment]
> Looking at the container logs I noticed, that services are started in parallel and even before cont-init.d scripts have finished. Race condition?

Actually, I wonder if this is down to the guid changed script that I didn't convert to a new s6 service (i actually forgot about this...) I'll look at converting it now and see where we get to

In the mean time, I've just pushed a change to `dev` that will allow one to set the `PH_VERBOSE` environment variable to `1`, and hopefully give us some more clues as to what is going on in these particular setups - I have so far not been able to repro any issues so it's a little difficult to debug


Those changes will be live once this finishes: https://github.com/pi-hole/docker-pi-hole/issues/1186

**Comment by PromoFaux** [comment]
Pushed some further changes to `dev` (currently building - will likely be built once this comment is 10 mins old) that hopefully sort some of the issues seen out.

Following @casperklein's suggestion, I have made `cron` the first service to be started, which depends on `base`, and then there is a hierarchy from there (no need to set all services' dependencies to `base`)

If there are still issues, then running with `PH_VERBOSE` set to `1` will hopefully help us further troubleshoot

**Comment by Nimon77** [comment]
I just try the new dev tag with `PH_VERBOSE`, the error is still there :
```
+ LIGHTTPD_CFG=lighttpd.conf.debian
+ installConfigs
+ printf '\n  %b Installing configs from %s...\n' '[i]' /etc/.pihole
+ version_check_dnsmasq
+ local dnsmasq_conf=/etc/dnsmasq.conf
+ local dnsmasq_conf_orig=/etc/dnsmasq.conf.orig
+ local dnsmasq_pihole_id_string=addn-hosts=/etc/pihole/gravity.list
  [i] Ensuring basic configuration by re-running select functions from basic-install.sh

  [i] Installing configs from /etc/.pihole...
+ local 'dnsmasq_pihole_id_string2=# Dnsmasq config for Pi-hole'\''s FTLDNS'
+ local dnsmasq_original_config=/etc/.pihole/advanced/dnsmasq.conf.original
+ local dnsmasq_pihole_01_source=/etc/.pihole/advanced/01-pihole.conf
+ local dnsmasq_pihole_01_target=/etc/dnsmasq.d/01-pihole.conf
+ local dnsmasq_rfc6761_06_source=/etc/.pihole/advanced/06-rfc6761.conf
+ local dnsmasq_rfc6761_06_target=/etc/dnsmasq.d/06-rfc6761.conf
+ [[ -f /etc/dnsmasq.conf ]]
+ printf '  %b Existing dnsmasq.conf found...' '[i]'
+ grep -q addn-hosts=/etc/pihole/gravity.list /etc/dnsmasq.conf
+ grep -q '# Dnsmasq config for Pi-hole'\''s FTLDNS' /etc/dnsmasq.conf
+ printf ' it is not a Pi-hole file, leaving alone!\n'
  [i] Existing dnsmasq.conf found... it is not a Pi-hole file, leaving alone!
+ printf '  %b Installing %s...' '[i]' /etc/dnsmasq.d/01-pihole.conf
+ [[ ! -d /etc/dnsmasq.d ]]
+ install -D -m 644 -T /etc/.pihole/advanced/01-pihole.conf /etc/dnsmasq.d/01-pihole.conf
+ printf '%b  %b Installed %s\n' '\r' '[✓]' /etc/dnsmasq.d/01-pihole.conf
+ sed -i s/@INT@// /etc/dnsmasq.d/01-pihole.conf
  [✓] Installed /etc/dnsmasq.d/01-pihole.conff...
+ [[ '' != '' ]]
+ sed -i '/^server=@DNS1@/d' /etc/dnsmasq.d/01-pihole.conf
+ [[ '' != '' ]]
+ sed -i '/^server=@DNS2@/d' /etc/dnsmasq.d/01-pihole.conf
s6-rc: fatal: timed out
+ sed -i s/@CACHE_SIZE@/10000/ /etc/dnsmasq.d/01-pihole.conf
s6-sudoc: fatal: unable to get exit status from server: Operation timed out
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
prog: fatal: stopping the container.
s6-rc: info: service _uid-gid-changer: stopping
s6-rc: info: service _uid-gid-changer successfully stopped
+ sed -i 's/^#conf-dir=\/etc\/dnsmasq.d$/conf-dir=\/etc\/dnsmasq.d/' /etc/dnsmasq.conf
s6-rc: info: service cron: stopping
+ [[ true == false ]]
+ sed -i 's/^#log-queries/log-queries/' /etc/dnsmasq.d/01-pihole.conf
Stopping cron
+ printf '  %b Installing %s...' '[i]' /etc/.pihole/advanced/06-rfc6761.conf
+ install -D -m 644 -T /etc/.pihole/advanced/06-rfc6761.conf /etc/dnsmasq.d/06-rfc6761.conf
  [✓] Installed /etc/dnsmasq.d/06-rfc6761.conf761.conf...
+ printf '%b  %b Installed %s\n' '\r' '[✓]' /etc/dnsmasq.d/06-rfc6761.conf
+ echo 'Google (ECS, DNSSEC);8.8.8.8;8.8.4.4;2001:4860:4860:0:0:0:0:8888;2001:4860:4860:0:0:0:0:8844
OpenDNS (ECS, DNSSEC);208.67.222.222;208.67.220.220;2620:119:35::35;2620:119:53::53
Level3;4.2.2.1;4.2.2.2;;
Comodo;8.26.56.26;8.20.247.20;;
DNS.WATCH (DNSSEC);84.200.69.80;84.200.70.40;2001:1608:10:25:0:0:1c04:b12f;2001:1608:10:25:0:0:9249:d69b
Quad9 (filtered, DNSSEC);9.9.9.9;149.112.112.112;2620:fe::fe;2620:fe::9
Quad9 (unfiltered, no DNSSEC);9.9.9.10;149.112.112.10;2620:fe::10;2620:fe::fe:10
Quad9 (filtered, ECS, DNSSEC);9.9.9.11;149.112.112.11;2620:fe::11;2620:fe::fe:11
Cloudflare (DNSSEC);1.1.1.1;1.0.0.1;2606:4700:4700::1111;2606:4700:4700::1001'
+ chmod 644 /etc/pihole/dns-servers.conf
+ [[ ! -r /etc/pihole/pihole-FTL.conf ]]
+ [[ ! -r /etc/pihole/custom.list ]]
+ install -T -m 0755 /etc/.pihole/advanced/Templates/pihole-FTL.service /etc/init.d/pihole-FTL
+ [[ true == true ]]
+ [[ ! -d /etc/lighttpd ]]
+ [[ -f /etc/lighttpd/lighttpd.conf ]]
+ mv /etc/lighttpd/lighttpd.conf /etc/lighttpd/lighttpd.conf.orig
s6-rc: info: service cron successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
+ install -D -m 644 -T /etc/.pihole/advanced/lighttpd.conf.debian /etc/lighttpd/lighttpd.conf
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
+ '[' '!' -f /etc/lighttpd/external.conf ']'
+ [[ -f /var/www/html/pihole/custom.php ]]
+ mkdir -p /run/lighttpd
+ chown www-data:www-data /run/lighttpd
+ mkdir -p /var/cache/lighttpd/compress
+ chown www-data:www-data /var/cache/lighttpd/compress
+ mkdir -p /var/cache/lighttpd/uploads
+ chown www-data:www-data /var/cache/lighttpd/uploads
+ installLogrotate
+ local 'str=Installing latest logrotate script'
+ local target=/etc/pihole/logrotate
+ printf '\n  %b %s...' '[i]' 'Installing latest logrotate script'
+ [[ -f /etc/pihole/logrotate ]]

+ grep -q /var/log/pihole.log /etc/pihole/logrotate
+ grep -q /var/log/pihole-FTL.log /etc/pihole/logrotate
+ printf '\n\t%b Existing logrotate file found. No changes made.\n' '[i]'
+ return 2
  [i] Installing latest logrotate script...
	[i] Existing logrotate file found. No changes made.
+ true
+ '[' '!' -f /etc/pihole/setupVars.conf ']'
+ set +e
+ mkdir -p /var/run/pihole /var/log/pihole
+ touch /var/log/pihole/FTL.log /var/log/pihole/pihole.log
+ chown pihole:root /etc/lighttpd
+ chown -R pihole:root /opt/pihole
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by PromoFaux** [comment]
@Nimon77 - Can you try setting and environment variable with the name: `S6_CMD_WAIT_FOR_SERVICES_MAXTIME` to a value of `0` to see if that makes a difference?

**Comment by PromoFaux** [comment]
> It works on my raspberry 4 (4Gb) but not on my raspberry 3 (1Gb),

I guess this is the clue... I'll build the image with that `S6_CMD_WAIT_FOR_SERVICES_MAXTIME ` variable already set to 0. All my testing has been done on faster hardware - I really should set up a zero or something for testing!

**Comment by PromoFaux** [comment]
@usma0118, @casperklein , @joe-eklund , @sleiner - Please try the above to see if that works for you, although I'm about to push a change to `dev` that includes this baked into the image

**Comment by arnold-alex** [comment]
> @usma0118, @casperklein , @joe-eklund , @sleiner - Please try the above to see if that works for you, although I'm about to push a change to `dev` that includes this baked into the image

This worked to fix the issue on my QNAP NAS.

**Comment by madzodiac** [comment]
Even still with latest release 2022.08.3 I get these log entries over and over again.
When using 2022.07.1 everything works perfectly fine.

"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:03.266195824Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:05:03.266469157Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:05:03.266509Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:03.267052487Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:05.309636253Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:05:05.30991297Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:05:05.31022625Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:05.312543115Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:07.360553834Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:05:07.360760188Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:05:07.362018255Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:07.364250484Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:09.335345769Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:05:09.335646184Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:05:09.338236537Z"}
{"log":"s6-supervise s6-linux-init-shutdownd: fatal: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:09.339996737Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:09.34131944Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:11.569470596Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:05:11.569694033Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:05:11.571981366Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:11.574089377Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:14.658955057Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:05:14.659159483Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:05:14.660385259Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:14.663281704Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:19.248281765Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:05:19.24848515Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:05:19.250391391Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:19.252531954Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:27.050841378Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:05:27.051058564Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:05:27.05109622Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:27.053291158Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:41.266843456Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:05:41.267096111Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:05:41.269320163Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:05:41.272063588Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:06:08.303466998Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:06:08.303714185Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:06:08.303749445Z"}
{"log":"s6-linux-init-shutdownd: fatal: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:06:08.309426398Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:06:08.322112643Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:07:01.748393488Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:07:01.748718122Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:07:01.75237571Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:07:01.754983354Z"}
{"log":"s6-svscan: warning: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:08:03.986092015Z"}
{"log":"s6-svscan: warning: executing into .s6-svscan/crash\n","stream":"stderr","time":"2022-08-29T18:08:03.986459773Z"}
{"log":"s6-supervise s6-linux-init-shutdownd: fatal: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:08:03.992410996Z"}
{"log":"s6-linux-init-shutdownd: fatal: unable to iopause: Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:08:03.992546933Z"}
{"log":"s6-svscan crashed. Killing everything and exiting.\n","stream":"stderr","time":"2022-08-29T18:08:03.996829518Z"}
{"log":"s6-linux-init-hpr: fatal: unable to reboot(): Operation not permitted\n","stream":"stderr","time":"2022-08-29T18:08:04.000432314Z"}

**Comment by casperklein** [comment]
> @usma0118, @casperklein , @joe-eklund , @sleiner - Please try the above to see if that works for you, although I'm about to push a change to `dev` that includes this baked into the image

Works for me. I could remove my custom patch 👍

**Comment by PromoFaux** [comment]
@madzodiac - yours looks to be an entirely different problem. Please open a new issue, and take care to fill out the requested information in the template so that we can better troubleshoot.

**Comment by usma0118** [comment]
Verified, `2022.08.03` is working fine.
> > @usma0118, @casperklein , @joe-eklund , @sleiner - Please try the above to see if that works for you, although I'm about to push a change to `dev` that includes this baked into the image
> 
> This worked to fix the issue on my QNAP NAS.

**Comment by sleiner** [comment]
> @usma0118, @casperklein , @joe-eklund , @sleiner - Please try the above to see if that works for you, although I'm about to push a change to `dev` that includes this baked into the image

same, `2022.8.3` works reliably - thanks so much! 🙌
